syntax = "proto3";

package qed.schemas.ai;

// A single, engine-validated action an NPC can take.
message CandidateOp {
  string op_name = 1; // e.g., "GATHER", "CRAFT"
  map<string, string> args = 2; // e.g., {"resource": "coal", "source_id": "bin_123"}
  string description = 3; // Human-readable description for the LLM
}

// The "briefing" sent from the engine to the AI service.
message AIDecisionRequest {
  string npc_id = 1;
  string archetype_id = 2;
  repeated string traits = 3;
  map<string, int32> inventory = 4; // A summary, e.g., {"iron_ore": 5}
  repeated CandidateOp candidates = 5;
  string decision_prompt = 6; // The prompt template from the archetype
}

// The "choice" sent back from the AI service to the engine.
message AIDecisionResponse {
  CandidateOp chosen_op = 1;
  string rationale = 2;
}


